src/RCDEF.CPP

```diff
@@ -609,27 +622,31 @@ int CDEC main( SI argc, char * argv[] )
 	/* check include directory argument if specified */
 	if (HFILESOUT)              /* if argv[6] not NUL */
 	{
-		getcwd( dbuff, 80);
+		dbuff = filesys::current_path();
 		incdir = argv[6];
-		if (chdir(incdir))
+		filesys::path incdirTemp = filesys::path(incdir);
+		filesys::current_path(incdirTemp, pathEC);
+		if (pathEC)
 		{
 			printf( "\nCannot find include directory '%s'.\n", incdir);
 			Errcount++;
 		}
-		chdir(dbuff);
+		filesys::current_path(dbuff);
 	}
 
 	/* check cpp file directory argument if specified */
 	if (CFILESOUT)              /* if argv[10] not NUL */
 	{
-		getcwd( dbuff, 80);
+		dbuff = filesys::current_path();
 		cFilesDir = argv[10];
-		if (chdir(cFilesDir))
+		filesys::path cFilesDirTemp = filesys::path(cFilesDir);
+		filesys::current_path(cFilesDirTemp, pathEC);
+		if (pathEC)
 		{
 			printf( "\nCannot find include directory '%s'.\n", cFilesDir);
 			Errcount++;
 		}
-		chdir(dbuff);
+		filesys::current_path(dbuff);
```

Comment:
Same here...though I wonder if we want to just make a function since this is nearly identical to the last set of changes.